<launch>
    <arg name="fcu_url" default="/dev/px4fmu"/>
    <arg name="gcs_url" default="tcp-l://0.0.0.0:5760"/> 

    <!-- mavros launch file configured for Jetson Nano -->
    <node pkg="mavros"
          type="mavros_node"
          name="mavros"
          required="false"
          clear_params="true"
          respawn="true"
          respawn_delay="1"
          output="screen">
        <param name="fcu_url" value="$(arg fcu_url)"/>
        <param name="gcs_url" value="$(arg gcs_url)"/>
        <param name="gcs_quiet_mode" value="true"/>
        <rosparam command="load" file="$(find coex_nano)/launch/mavros_config.yaml"/>
        <remap from="mavros/distance_sensor/rangefinder_sub" to="rangefinder/range"/>
        <remap from="mavros/obstacle_distance/send" to="depth_camera/laserscan_transform/transformed_laserscan"/>
        <rosparam param="plugin_whitelist">
            - altitude
            - command
            - distance_sensor
            - ftp
            - global_position
            - imu
            - local_position
            - manual_control
            # - mocap_pose_estimate
            - obstacle_distance
            - odom
            - param
            - px4flow
            - rc_io
            - setpoint_attitude
            - setpoint_position
            - setpoint_raw
            - setpoint_velocity
            - sys_status
            - sys_time
            - vision_pose_estimate
            # - vision_speed_estimate
            # - waypoint
            </rosparam>
    </node>

    <rosparam param="rangefinder/range">
        subscriber: true
        id: 1
        orientation: PITCH_270
        covariance: 1  # cm
    </rosparam>

    <!-- Copter visualization -->
    <node name="visualization" pkg="mavros_extras" type="visualization">
            <remap to="mavros/local_position/pose" from="local_position"/>
            <remap to="mavros/setpoint_position/local" from="local_setpoint"/>
            <param name="fixed_frame_id" value="map"/>
            <param name="child_frame_id" value="base_link"/>
            <param name="marker_scale" value="1"/>
            <param name="max_track_size" value="20"/>
            <param name="num_rotors" value="4"/>
    </node>
</launch>
